pandas/tests/reshape/test_cut.py

```diff
@@ -112,6 +112,27 @@ def test_bins_not_monotonic():
         cut(data, [0.1, 1.5, 1, 10])
 
 
+@pytest.mark.parametrize("x, bins, expected", [
+    (date_range("2017-12-31", periods=3),
+     [Timestamp.min, Timestamp('2018-01-01'), Timestamp.max],
+     np.array([0, 0, 1], dtype="int8")),
+    ([-1, 0, 1],
+     np.array([np.iinfo(np.int64).min, 0, np.iinfo(np.int64).max],
+              dtype="int64"),
+     np.array([0, 0, 1], dtype="int8")),
+    ([np.timedelta64(-1), np.timedelta64(0), np.timedelta64(1)],
+     np.array([
+         np.timedelta64(-np.iinfo(np.int64).max),
+         np.timedelta64(0),
+         np.timedelta64(np.iinfo(np.int64).max)]),
+     np.array([0, 0, 1], dtype="int8")),
+
+])
+def test_bins_monotic_not_overflowing(x, bins, expected):
+    result = cut(x, bins)
+    tm.assert_numpy_array_equal(result.codes, expected)
```

Comment:
can you use assert_index_equal here instead (you need to construct the expected) in the test itself (or in the parameterize is ok too)

Reply:
is this what you meant? https://github.com/pandas-dev/pandas/pull/26063/files#diff-8b8294096bd72eaa8a56c7e2fa7473d5R118